extensions/flyway/runtime/src/main/java/io/quarkus/flyway/runtime/FlywayCreator.java

```diff
@@ -33,7 +33,14 @@ public FlywayCreator withCallbacks(Collection<Callback> callbacks) {
 
     public Flyway createFlyway(DataSource dataSource) {
         FluentConfiguration configure = Flyway.configure();
-        configure.dataSource(dataSource);
+
+        if (flywayRuntimeConfig.username.isPresent() && flywayRuntimeConfig.password.isPresent()
+                && flywayRuntimeConfig.jdbcUrl.isPresent()) {
+            configure.dataSource(flywayRuntimeConfig.jdbcUrl.get(), flywayRuntimeConfig.username.get(),
+                    flywayRuntimeConfig.password.get());
```

Comment:
This assumes that all three have been set. We should warn (or perhaps even fail) if at least one but not all three are set.
Or perhaps the URL does not need to be set but can be taken from the corresponding DataSource.